trie/patriciaMerkleTrie.go

```diff
@@ -621,6 +621,29 @@ func (tr *patriciaMerkleTrie) Close() error {
 	return nil
 }
 
+// MarkStorerAsSyncedAndActive marks the storage as synced and active
+func (tr *patriciaMerkleTrie) MarkStorerAsSyncedAndActive() {
+	epoch, err := tr.trieStorage.GetLatestStorageEpoch()
+	if err != nil {
+		log.Error("getLatestStorageEpoch error", "error", err)
+	}
+
+	err = tr.trieStorage.Put([]byte(common.TrieSyncedKey), []byte(common.TrieSyncedVal))
+	if err != nil {
+		log.Error("error while putting trieSynced value into main storer after sync", "error", err)
+	}
+
+	lastEpoch := epoch - 1
+	if epoch == 0 {
+		lastEpoch = 0
+	}
+
+	err = tr.trieStorage.PutInEpoch([]byte(common.ActiveDBKey), []byte(common.ActiveDBVal), lastEpoch)
```

Comment:
when we finish the snapshot we call PutInEpochWithoutCache for writing on the ActiveDBKey. Is this correct?

Reply:
When we get the value for ActiveDBKey we ignore the cache, so it does not really matter which Put is used. Will change for consistency.